Lib/copy.py

```diff
@@ -84,22 +84,19 @@ def copy(x):
         return _copy_immutable(x)
 
     copier = getattr(cls, "__copy__", None)
-    if copier:
+    if copier is not None:
         return copier(x)
 
     reductor = dispatch_table.get(cls)
-    if reductor:
+    if reductor is not None:
         rv = reductor(x)
     else:
         reductor = getattr(x, "__reduce_ex__", None)
-        if reductor:
+        if reductor is not None:
+            # note that object.__reduce_ex__ calls __reduce__
```

Comment:
There is small behavior change. Currently if set `__reduce_ex__` to `None`, `copy()` falls back to `__reduce__`. Patched code raises `Error`.

This change looks not bad to me.

Reply:
Added a test for this behavior.